tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,293 @@
+    record_info('Guest Power Off');
+    $powerops     = 'power.off';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
+}
+
+sub do_networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+    my ($powerops, $powerops_ret);
+
+    record_info('Networking Tests');
+    # Boot up the VM if it's powered off
+    $powerops     = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    login_vm_console();
+
+    # Test network via assigned IPv4 address
+    assert_script_run("ping -I $vm_ip -4 -c3 openqa.suse.de");
+    assert_script_run("ping -I $vm_ip -4 -c3 www.suse.com");
+}
+
+sub do_clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($powerops, $powerops_ret, $g_init_time, $diff_secs);
+
+    record_info('Clock Sync Tests');
+    # Boot up the VM if it's powered off
+    $powerops     = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    select_console('sut') if (current_console() ne 'sut');
+
+    # Set the last day of host time in guest before enabling timesync service
+    $g_init_time = init_guest_time();
+    record_info('Guest current time before time sync enabled', $g_init_time);
+
+    # Enable timesync service in guest VM
+    if (script_output('vmware-toolbox-cmd timesync status', proceed_on_failure => 1, timeout => 90) eq 'Disabled') {
+        assert_script_run('vmware-toolbox-cmd timesync enable', timeout => 90);
+    }
+
+    # Check the time difference between host and guest after timesync service enabled
+    $diff_secs = get_diff_seconds();
+
+    if ($diff_secs < 10) {
+        record_info('Clock synchronization is successful.');
+    } else {
+        die 'Clock synchronization failed.';
+    }
+
+    # Disable clock sync tests
+    select_console('svirt');
+
+    $powerops     = 'power.shutdown';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
+
+    disable_vm_time_synchronization($vm_name);
+
+    $powerops     = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    login_vm_console();
+
+    # Set guest time with a given time which different from host time
+    $g_init_time = init_guest_time();
+    record_info('Set guest time after time sync disabled', $g_init_time);
+
+    # Guest time will not be synced up with host after timesync service disabled
+    $diff_secs = get_diff_seconds();
+
+    if ($diff_secs > 86420) {
```

Comment:
@nanzhg This diff_secs should be within a range, sth like 86400 < diff_secs < 86420?

Reply:
Corrected. Thanks for pointing out. Will update the verification run once it's done.